validate/validate.go

```diff
@@ -356,9 +356,24 @@ func (v *Validator) CheckLinux() (msgs []string) {
 		msgs = append(msgs, fmt.Sprintf("On Linux, hostname requires a new UTS namespace to be specified as well"))
 	}
 
+	// Linux devices validation
+	devList := make(map[string]bool)
+	typeList := make(map[string]string)
 	for index := 0; index < len(v.spec.Linux.Devices); index++ {
-		if !deviceValid(v.spec.Linux.Devices[index]) {
-			msgs = append(msgs, fmt.Sprintf("device %v is invalid.", v.spec.Linux.Devices[index]))
+		device := v.spec.Linux.Devices[index]
+		if !deviceValid(device) {
+			msgs = append(msgs, fmt.Sprintf("device %v is invalid.", device))
+		}
+		if _, exists := devList[device.Path]; exists {
+			msgs = append(msgs, fmt.Sprintf("device %s is duplicated", device.Path))
+		} else {
+			devList[device.Path] = true
+		}
+		devId := fmt.Sprintf("%d:%d", device.Major, device.Minor)
+		if existId, exists := typeList[device.Type]; exists && existId == devId {
```

Comment:
`typeList` can also be a `map[string]bool` (used as a set, so the value doesn't matter).  `devId` should be the `typeList` key and contain the type as well (normalizing `u` -> `c`).